In [28]:
import pandas as pd
import os

df = pd.read_json('../Data/Serie_A_2014_squads.json')

df = df.T
df.head(3)

0  \
1  {'Home': {'Squad': ['141698', '137644', '14474...   
2  {'Home': {'Squad': ['137990', '172996', '16795...   
3  {'Home': {'Squad': ['107639', '161940', '45966...   

                                                   1  \
1  {'Home': {'Squad': ['141698', '137644', '14474...   
2  {'Home': {'Squad': ['137990', '172996', '16795...   
3  {'Home': {'Squad': ['107639', '161940', '45966...   

                                                   2  \
1  {'Home': {'Squad': ['141698', '137644', '14474...   
2  {'Home': {'Squad': ['137990', '172996', '16795...   
3  {'Home': {'Squad': ['107639', '161940', '45966...   

                                                   3  \
1  {'Home': {'Squad': ['141698', '137644', '14474...   
2  {'Home': {'Squad': ['137990', '172996', '16795...   
3  {'Home': {'Squad': ['107639', '161940', '31664...   

                                                   4  \
1  {'Home': {'Squad': ['141698', '137644', '14474...   
2  {'Home': {'Squad': ['137990', '172996', '16795...   
3                                                NaN   

                                                   5    6    7    8  
1  {'Home': {'Squad': ['141698', '137644', '14474...  NaN  NaN  NaN  
2  {'Home': {'Squad': ['137990', '172996', '16795...  NaN  NaN  NaN  
3                                                NaN  NaN  NaN  NaN

Vamos pegar as informações dos ID's dos jogadores com as informações dos Squads. Começaremos pelo campeonato da Série A de 2013 até 2024, e nossa intenção é definir um rating inicial de 1500 para cada jogador.

Com isso, conseguiremos calcular um rating de uma partida e atualizar o rating dos jogadores após ela nesse dataframe com os ID's.

In [7]:

def collect_unique_ids(data_folder, file_names, years):
    """
    Função para coletar IDs únicos de jogadores em arquivos JSON.

    :param data_folder: Caminho para o diretório onde os arquivos JSON estão localizados.
    :param file_names: Lista de nomes de arquivos base (ex: ['Serie_A', 'Serie_B', 'Copa_do_Brasil']).
    :param years: Lista ou range de anos dos arquivos a serem processados.
    :return: Lista de IDs únicos coletados de todos os arquivos.
    """

    unique_ids = set()

    def process_dataframe(df):

        for column in df.columns:
            for cell in df[column]:
                if isinstance(cell, dict):  
                    for key in ['Home', 'Away']:
                        team_info = cell.get(key, {})
                        # Adicionar os IDs do 'Squad'
                        squad_ids = team_info.get('Squad', [])
                        unique_ids.update(squad_ids)


    for file_name in file_names:
        for year in years:
            file_path = os.path.join(data_folder, f'{file_name}_{year}_squads.json')
            if os.path.exists(file_path):
                df = pd.read_json(file_path)
                process_dataframe(df)
            else:
                print(f'Arquivo não encontrado: {file_path}')


    return list(unique_ids)

data_folder = '../Data/'
#file_names = ['Serie_A', 'Serie_B', 'Serie_C', 'Serie_D', 'CdB']
file_names = ['Serie_A']
years = range(2013, 2025)

unique_player_ids = collect_unique_ids(data_folder, file_names, years)

print(f'Total de IDs únicos coletados: {len(unique_player_ids)}')

Total de IDs únicos coletados: 3157


Total de 15448 diferentes jogadores registrados!

Vamos trabalhar com os 3157 da série A. Simularemos inicialmente idades e jogos jogados.

In [11]:
df_players = pd.DataFrame(unique_player_ids, columns=['player'])

# Adicionar a coluna 'rating' com valor inicial de 1500 para cada jogador
df_players['rating'] = 1500
df_players['age'] = 25
df_players['games_played'] = 50

df_players.to_excel('id_players.xlsx')

# Mostrar o DataFrame criado
df_players.head()

player  rating  age  games_played
0  553597    1500   25            50
1  155516    1500   25            50
2  554744    1500   25            50
3  313780    1500   25            50
4  292114    1500   25            50